lib/salesforce/donation_creator.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'salesforce/client_api'
+require 'salesforce/donation_validator'
+
+module Salesforce
+  class DonationCreator
+    def self.execute(data, supporter, client = ClientAPI.new)
+      new(client, data, supporter).execute
+    end
+
+    def initialize(client, data, supporter)
+      @client = client
+      @data = data
+      @supporter = supporter
+    end
+
+    def execute
+      Result.new(donation, errors)
+    end
+
+    private
+
+    attr_reader :client, :data, :supporter
+
+    def table
+      'Opportunity'
+    end
+
+    def donation
```

Comment:
```
return unless validation.okay?
fetch(create)
```

Reply:
no, because create might return nil if there is any error with the client, and then fetch will blow up.